slack-plugin/src/main/kotlin/slack/gradle/lint/LintTasks.kt

```diff
@@ -50,26 +66,178 @@ internal object LintTasks {
+                slackProperties,
+                affectedProjects,
+                onProjectSkipped
+              )
+            }
+        }
+      }
+    project.plugins.all(projectAllAction)
+  }
+
+  /** Android Lint configuration entry point for Android projects. */
+  private fun Project.configureAndroidProjectForLint(
+    slackProperties: SlackProperties,
+    affectedProjects: Set<String>?,
+    onProjectSkipped: (String, String) -> Unit,
+  ) =
+    androidExtension.finalizeDsl { extension ->
+      project.log("Applying ciLint to Android project")
+
+      project.log(
+        "Configuring android lint tasks. isApp=${extension is ApplicationAndroidComponentsExtension}"
+      )
+
+      project.log("Creating ciLint task")
+      val ciLintTask =
+        project.tasks.register(CI_LINT_TASK_NAME) { group = LifecycleBasePlugin.VERIFICATION_GROUP }
+
+      val ciLintVariants = slackProperties.ciLintVariants
+      if (ciLintVariants != null) {
+        ciLintTask.configure {
+          // Even if the task isn't created yet, we can do this by name alone and it will resolve at
+          // task configuration time.
+          ciLintVariants.splitToSequence(',').forEach { variant ->
+            logger.debug("Using variant $variant for ciLint task")
+            val lintTaskName = "lint${variant.capitalizeUS()}"
+            dependsOn(lintTaskName)
+          }
+        }
+      } else {
+        androidExtension.onVariants { variant ->
+          val lintTaskName = "lint${variant.name.capitalizeUS()}"
+          project.log("Adding $lintTaskName to ciLint task")
+          ciLintTask.configure {
+            // Even if the task isn't created yet, we can do this by name alone and it will resolve
+            // at
+            // task configuration time.
+            dependsOn(lintTaskName)
+          }
+        }
+      }
+
+      configureLint(
+        extension.lint,
+        ciLintTask,
+        slackProperties,
+        affectedProjects,
+        onProjectSkipped,
+        slackProperties.requireAndroidSdkProperties()
+      )
+    }
+
+  /** Android Lint configuration entry point for non-Android projects. */
+  private fun Project.configureNonAndroidProjectForLint(
+    slackProperties: SlackProperties,
+    affectedProjects: Set<String>?,
+    onProjectSkipped: (String, String) -> Unit,
+  ) = afterEvaluate {
+    // The lint plugin expects certain configurations and source sets which are only added by
+    // the Java and Android plugins. If this is a multiplatform project targeting JVM, we'll
+    // need to manually create these configurations and source sets based on their multiplatform
+    // JVM equivalents.
+    addSourceSetsForMultiplatformAfterEvaluate()
+
+    // For Android projects, the Android Gradle Plugin is responsible for applying the lint plugin;
+    // however, we need to apply it ourselves for non-Android projects.
+    apply(mapOf("plugin" to "com.android.lint"))
```

Comment:
not a big deal, but it might be simpler to use `pluginManager.apply("com.android.lint")`.

Reply:
ah true

Reply:
[72408f9](https://github.com/slackhq/slack-gradle-plugin/pull/612/commits/72408f90c1ee065df85acca479e95b624e9edb84)